### PR TITLE
Expose passing of MapboxNavigationOptions to NavigationView

### PR DESCRIPTION
**Pull request by danesfeder**
- Will allow developers to pass their own customized `MapboxNavigationOptions` to `NavigationView` via `NavigationViewOptions`.  

- `MapboxNavigation` will then be initialized with these options at the start of the UI.

Note: Metric support should land first - I broke into two separate PRs because this PR changes how we initialize the UI and should be tested more thoroughly (like with rotation).

cc @ericrwolfe

### CODE REVIEW

**Comment by Danny-James** [comment]
Could you provide some sample code on how I would create the NavigationOptions please?

**Comment by danesfeder** [comment]
@cammace this is ready for 👀

File: app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java [hunk 1]
```diff
@@ -272,9 +270,7 @@ private void launchNavigationWithRoute() {
   }
 
   private boolean validRouteResponse(Response<DirectionsResponse> response) {
-    return response.body() != null
-      && response.body().routes() != null
-      && response.body().routes().size() > 0;
+    return response.body() != null && response.body().routes().size() > 0;
```
Comment (cammace) [inline app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java]: `&& response.body().routes().size() > 0;` to `&& !response.body().routes().isEmpty();`

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java [hunk 2]
```diff
@@ -270,6 +268,14 @@ public void finishNavigationView() {
    * @param options with containing route / coordinate data
    */
   public void startNavigation(NavigationViewOptions options) {
+    // Initialize navigation with options from NavigationViewOptions
+    navigationViewModel.initializeNavigationOptions(getContext().getApplicationContext(),
+      options.navigationOptions().toBuilder().isFromNavigationUi(true).build());
```
Comment (cammace) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: Is it possible for `options.navigationOptions()` to be null?

Comment (danesfeder) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: The `NavigationViewOptions` `builder()` will always provide `MapboxNavigationOptions` with the defaults - so someone would have to explicitly set it to null

**Comment by danesfeder** [comment]
@cammace ready for another round here 👀

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java (hunk 2 above)
Comment (cammace) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: Annotate the parameter with `@NonNull` then

**Pull request merged**